### PR TITLE
JAMES-1823 Register QuotaUpdate listener only one time

### PR DESCRIPTION
**Pull request by aduprat**

### CODE REVIEW

File: mailbox/store/src/test/java/org/apache/james/mailbox/store/StoreMailboxManagerTest.java [hunk 1]
```diff
@@ -159,5 +163,24 @@ public void getMailboxShouldThrowWhenMailboxDoesNotMatchUser() throws Exception
...
+    @Test
+    public void quotaUpdaterShouldBeInitializeOnlyOneTime() throws Exception {
+        MyDelegatingMailboxListener delegatingListener = new MyDelegatingMailboxListener();
+        storeMailboxManager.setDelegatingMailboxListener(delegatingListener);
+        storeMailboxManager.init();
+        int expectedListenersNumber = delegatingListener.getListeners().size();
+
+        storeMailboxManager.init();
+
+        assertThat(delegatingListener.getListeners().size()).isEqualTo(expectedListenersNumber);
```
Comment (mbaechler) [inline mailbox/store/src/test/java/org/apache/james/mailbox/store/StoreMailboxManagerTest.java]: hasSize

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/event/DefaultDelegatingMailboxListener.java [hunk 2]
```diff
@@ -107,4 +107,13 @@ protected void deliverEventToGlobalListeners(Event event) {
         }
     }
 
+    @Override
+    public boolean isListeningTo(MailboxListener mailboxListener) {
```
Comment (mbaechler) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/event/DefaultDelegatingMailboxListener.java]: I don't understand this method : `this` is a listener and you ask if it listens another listener.
I mean, a listener usually listen to events, don't you have a typing problem here ?

**Comment by aduprat** [comment]
This bug only arise when deploying with spring, and there's no integration test for that...
I don't know if i had to spend 2 days building integration tests....

This happens because all mailbox implementations are packaged in spring.

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/event/DefaultDelegatingMailboxListener.java (hunk 2 above)
Comment (chibenwa) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/event/DefaultDelegatingMailboxListener.java]: It is a class notified after event creation, that takes care of notifying other listeners about events.

Comment (mbaechler) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/event/DefaultDelegatingMailboxListener.java]: Still a listener is not listening to another listener, isn't it ?

Comment (aduprat) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/event/DefaultDelegatingMailboxListener.java]: Ok, i've changed the method name.

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/event/DefaultDelegatingMailboxListener.java [hunk 3]
```diff
@@ -107,4 +107,13 @@ protected void deliverEventToGlobalListeners(Event event) {
         }
     }
 
+    @Override
+    public boolean isRegistered(MailboxListener mailboxListener) {
+        return registry.isListeningTo(mailboxListener);
```
Comment (mbaechler) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/event/DefaultDelegatingMailboxListener.java]: registry is listening to mailboxListener ? It's still not ok

File: mailbox/store/src/test/java/org/apache/james/mailbox/store/StoreMailboxManagerTest.java (hunk 1 above)
Comment (mbaechler) [inline mailbox/store/src/test/java/org/apache/james/mailbox/store/StoreMailboxManagerTest.java]: what about my comment ?

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java [hunk 4]
```diff
@@ -195,7 +195,7 @@ public void init() throws MailboxException {
         if (quotaRootResolver == null) {
             quotaRootResolver = new DefaultQuotaRootResolver(mailboxSessionMapperFactory);
         }
-        if (quotaUpdater != null && quotaUpdater instanceof MailboxListener) {
+        if (shouldInitQuotaUpdaterListener()) {
             this.addGlobalListener((MailboxListener) quotaUpdater, null);
```
Comment (mbaechler) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java]: why it's not the registry that handle that logic ?

**Comment by chibenwa** [comment]
Quynh just merged related changes :-)

**Pull request closed without merging**